File: 0035-search-insert-position/0035-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int n = nums.length;
        int l = 0;
        int h = n - 1;
        int mid = (l + h)/2;

        while(l <= h){
            if(target == nums[mid])
                return mid;
            else if(target > nums[mid])
                l = mid + 1;
            else
                h = mid - 1;
            
            mid = (l + h)/2;
        }

        return l;
    }
}

File: 0035-search-insert-position/0035-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        n = len(nums)
        l = 0
        h = n - 1

        while l<=h:
            mid = (l + h) // 2
            x = nums[mid]

            if target == x:
                return mid
            elif target<x:
                h = mid - 1
            else:
                l = mid + 1

        
        return l
        
            


